llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -729,6 +769,29 @@ def tensor_hook(
                         monitor_process = None
 
                     # Spawn a new process to register the model.
+                    # Slower method to register the model via log_model.
+                    if self.use_mlflow_log_model:
```

Comment:
If `self.use_mlflow_log_model` is true, do we need to both log and register the model? Can we only log to avoid duplicate artifact uploads?

Reply:
We do need to log and register because that's the only way to resolve the infinitely pending model registration issues we've been debugging for Azure PL support. [Debugging context is here](https://docs.google.com/document/d/1fh38Zw_-0qrVGuwUis0RgjJ8acli2_Lk2Q6bH0PvIDw/edit#heading=h.mpoax8wdsew0).